docs/source/mar_specification.rst

```diff
@@ -0,0 +1,137 @@
+::
+  ModelName
+  ┣━ configs
+  ┃  ┗━ metadata.json
+  ┣━ models
+  ┃  ┣━ model.pt
+  ┃  ┗━ model.ts
+  ┗━ docs
+     ┣━ README.md
+     ┗━ license.txt
+
+
+These files mostly are required to be present with the given names for the directory to define a valid MAR:
+
+* **metadata.json**: netadata information in JSON format relating to the type of model, definition of input and output tensors, versions of the model and used software, and other information described below.
+* **model.pt**: the state dictionary of a saved model, the information to instantiate the model must be found in the metadata file.
+* **model.ts**: the Torchscript saved model if the model is compatible with being saved correctly in this format.
+* **README.md**: plain-language information on the model, how to use it, author information, etc. in Markdown format.
+* **license.txt**: software license attached to the model, can be left blank if no license needed.
+
+Archive Format
+==============
+
+The MAR directory and its contents can be compressed into a zip file to constitute a single file package. When unzipped into a directory this file will reproduce the above directory structure, and should itself also be named after the model it contains.
+
+The Torchscript file format is also just a zip file with a specific structure. When creating such an archive with `save_net_with_metadata` a MAR-compliant Torchscript file can be created by including the contents of `metadata.json` as the `meta_values` argument of the function, and other files included as `more_extra_files` entries. These will be stored in a `extras` directory in the zip file and can be retrieved with `load_net_with_metadata` or with any other library/tool that can read zip data. In this format the `model.*` files are obviously not needed by `README.md` and `license.txt` can be added as more extra files.
+
+metadata.json File
+==================
+
+This file contains the metadata information relating to the model, including what the shape and format of inputs and outputs are, what the meaning of the outputs are, what type of model is present, and other information. The JSON structure is a dictionary containing a defined set of keys with additional user-specified keys. The mandatory keys are as follows:
+
+* **version**: version of the stored model.
+* **monai_version**: version of MONAI the MAR was generated on, later versions expected to work.
+* **pytorch_version**: version of Pytorch the MAR was generated on, later versions expected to work.
+* **numpy_version**: version of Numpy the MAR was generated on, later versions expected to work.
+* **optional_packages_version**: dictionary relating optional package names to their versions, these packages are not needed but are recommended to be isntalled with this stated minimum version.
```

Comment:
```suggestion
* **optional_packages_version**: dictionary relating optional package names to their versions, these packages are not needed but are recommended to be installed with this stated minimum version.
```

Reply:
Also what about required dependencies (apart from torch, monai, numpy)?

Reply:
Those would go into `optional_packages_version` so maybe call this `other_package_versions` instead.

Reply:
Hi @ericspod , I would suggest `optional_packages_version` or `optional_dependencies_version`, to be consistent with our installation guide doc:
https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

Thanks.